File: php/src/Exceptions/UcotronServerException.php
<?php

declare(strict_types=1);

namespace Ucotron\Sdk\Exceptions;

class UcotronServerException extends UcotronException
{
    public function __construct(
        public readonly int $statusCode,
        public readonly ?string $errorBody,
        string $message = '',
        ?\Throwable $previous = null,
    ) {
        $msg = $message ?: "Server returned HTTP {$statusCode}";
        if ($errorBody !== null) {
            $msg .= ": {$errorBody}";
        }
        parent::__construct($msg, $statusCode, $previous);
    }
}

File: java/ucotron-sdk/src/main/java/com/ucotron/sdk/UcotronServerException.java
package com.ucotron.sdk;

/**
 * Exception for server-side errors (4xx and 5xx HTTP responses).
 */
public class UcotronServerException extends UcotronException {
    private final int statusCode;

    public UcotronServerException(int statusCode, String message) {
        super("HTTP " + statusCode + ": " + message);
        this.statusCode = statusCode;
    }

    public int getStatusCode() { return statusCode; }
}
